lib/src/mock_storage_reference.dart

```diff
@@ -5,31 +5,31 @@ import 'package:firebase_storage/firebase_storage.dart';
 import 'package:firebase_storage_mocks/firebase_storage_mocks.dart';
 import 'package:mockito/mockito.dart';
 
-class MockStorageReference extends Mock implements StorageReference {
+class MockReference extends Mock implements Reference {
   final MockFirebaseStorage _storage;
   final String _path;
-  final Map<String, MockStorageReference> children = {};
+  final Map<String, MockReference> children = {};
 
-  MockStorageReference(this._storage, [this._path = '']);
+  MockReference(this._storage, [this._path = '']);
 
   @override
-  StorageReference child(String path) {
+  Reference child(String path) {
     if (!children.containsKey(path)) {
-      children[path] = MockStorageReference(_storage, '$_path/$path');
+      children[path] = MockReference(_storage, '$_path$path');
     }
-    return children[path];
+    return children[path] as Reference;
```

Comment:
If you were trying to fix this error:
> A value of type 'MockReference?' can't be returned from the method 'child' because it has a return type of 'Reference'.

The better way is to declare it as non-null: `return children[path]!;`. A MockReference is a Reference, so issue is not with the type but with the nullability.